File: sdk/node/examples/audit_query.js
/**
 * Audit query example — list logs, get stats, export CSV.
 *
 * Setup:
 *   npm install wrapsec-node
 *   export WRAPSEC_API_KEY=wsk_live_...
 *
 * Run:
 *   node examples/audit_query.js
 */

const fs      = require("fs")
const WrapSec = require("wrapsec-node").default

const client = new WrapSec({ apiKey: process.env.WRAPSEC_API_KEY })

async function main() {
  // Recent blocked requests
  console.log("=== Recent BLOCKs ===")
  const logs = await client.auditList({ decision: "BLOCK", limit: 5 })
  for (const log of logs) {
    console.log(`  ${log.traceId}  ${log.primaryReason.padEnd(35)}  conf=${log.confidence.toFixed(2)}  ${log.createdAt.slice(0, 19)}`)
  }
  if (!logs.length) console.log("  No blocked requests yet.")

  // Stats
  console.log("\n=== Stats ===")
  const stats = await client.auditStats()
  console.log(`  Total:     ${stats.totalRequests.toLocaleString()}`)
  if (stats.totalRequests) {
    console.log(`  Blocked:   ${stats.blockCount.toLocaleString()}  (${(stats.blockRate * 100).toFixed(1)}%)`)
    console.log(`  Sanitized: ${stats.sanitizeCount.toLocaleString()}`)
    console.log(`  Allowed:   ${stats.allowCount.toLocaleString()}`)
    console.log(`  Avg latency:  ${stats.avgLatencyMs.toFixed(1)}ms`)
    console.log(`  P95 latency:  ${stats.p95LatencyMs.toFixed(1)}ms`)
    const sev = stats.severityCounts
    console.log(`  Severity:  CRITICAL=${sev.CRITICAL}  HIGH=${sev.HIGH}  MEDIUM=${sev.MEDIUM}  LOW=${sev.LOW}`)
  }

  // Export CSV
  console.log("\n=== Export ===")
  const csv = await client.auditExport({ limit: 100 })
  fs.writeFileSync("audit_export.csv", csv)
  console.log(`  Exported ${csv.length.toLocaleString()} bytes to audit_export.csv`)
}

main().catch(err => {
  console.error(err.message)
  process.exit(1)
})

File: sdk/python/examples/audit_query.py
"""
Audit query example - list logs, get stats, export CSV.

Setup:
    pip install wrapsec-python
    export WRAPSEC_API_KEY=wsk_live_...

Run:
    python examples/audit_query.py
"""

from pathlib import Path
import wrapsec

client = wrapsec.Client()

# Recent blocked requests
print("=== Recent BLOCKs ===")
logs = client.audit_list(decision="BLOCK", limit=5)
for log in logs:
    print(f"  {log.trace_id}  {log.primary_reason:<35}  conf={log.confidence:.2f}  {log.created_at[:19]}")

# Stats for all time
print("\n=== Stats ===")
stats = client.audit_stats()
total = stats.total_requests
if total:
    print(f"  Total:     {total:,}")
    print(f"  Blocked:   {stats.block_count:,}  ({stats.block_rate * 100:.1f}%)")
    print(f"  Sanitized: {stats.sanitize_count:,}")
    print(f"  Allowed:   {stats.allow_count:,}")
    print(f"  Avg latency:  {stats.avg_latency_ms:.1f}ms")
    print(f"  P95 latency:  {stats.p95_latency_ms:.1f}ms")
    sev = stats.severity_counts
    print(f"  Severity:  CRITICAL={sev.get('CRITICAL', 0)}  HIGH={sev.get('HIGH', 0)}  MEDIUM={sev.get('MEDIUM', 0)}  LOW={sev.get('LOW', 0)}")
else:
    print("  No requests yet.")

# Export CSV
print("\n=== Export ===")
csv_bytes = client.audit_export(limit=100)
out = Path("audit_export.csv")
out.write_bytes(csv_bytes)
print(f"  Exported {len(csv_bytes):,} bytes to {out}")
